sys/shell/shell.c

```diff
@@ -246,23 +276,32 @@ static int readline(char *buf, size_t size)
         /* DOS newlines are handled like hitting enter twice, but empty lines are ignored. */
         /* Ctrl-C cancels the current line. */
         if (c == '\r' || c == '\n' || c == ETX) {
-            *line_buf_ptr = '\0';
+            if (c == ETX) {
+                curr_pos = 0;
+                length_exceeded = 0;
```

Comment:
/0/false/

Reply:
this flag will be gone in  #12099

Reply:
that might be, but the flag is introduces in this PR as `bool` and below it is set to `true` so why not use the (more) appropriate `false` here?